core-rust/state-manager/src/mempool/pending_transaction_result_cache.rs

```diff
@@ -339,6 +338,49 @@ pub enum RetryFrom {
     Whenever,
 }
 
+impl RetryFrom {
+    fn exponential_backoff(
+        from: SystemTime,
+        base_delay: Duration,
+        exponent: f32,
+        multiplier_range: core::ops::Range<f32>,
+    ) -> Self {
+        Self::calculated_or_never(|| {
+            let multiplier: f32 = 2.0_f32.powf(exponent).clamp(multiplier_range.start, multiplier_range.end);
```

Comment:
maybe better to clamp exponent instead?

Reply:
Out of interest, why do you think it would be better?

I actually originally had a clamp of the exponent, but I changed to clamping the multiplier as it felt more declarative / easier to understand; but maybe that assumption is wrong! I'm happy to change back though if you prefer?

For what it's worth, `(2^100) = +Inf`, which clamps to `multiplier_range.end` - will add some unit tests for this though.